app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorViewModel.kt

```diff
@@ -0,0 +1,126 @@
+package com.example.sharewhatyoucanproject
+
+import android.content.Context
+import android.net.Uri
+import android.os.Looper
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.utils.showToast
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.GeoPoint
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.storage.UploadTask
+import java.util.*
+
+class DonorViewModel(
+    val context: Context?,
+    var db: FirebaseFirestore?,
+    var storageReference: StorageReference?,
+) : ViewModel() {
```

Comment:
if you extend `AndroidViewModel` instead of `ViewModel` here then you should have a reference to the application context. **Do not pass any other type of context to a view model** as that will cause memory leaks. View models are not destroyed when an activity is destroyed during a configuration change. Activities do not get cleaned up until all references to the activity are destroyed as well. Since the View model does not get destroyed, and it contains a hard reference to the activity, that means the activity will never be released, and you will get multiple instances in the stack.